### PR TITLE
JBPM-8901: Stunner: Multiple issues on node selection

### PR DESCRIPTION
**Pull request by handreyrc**
@romartin, @inodeman Could you please review this PR?

### CODE REVIEW

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Assigning the review to @LuboTerifaj as we agreed last week, he'll checks delegates to @domhanak ...

**Comment by LuboTerifaj** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
@LuboTerifaj @inodeman artifacts generated - ready for review. Thanks!

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/test/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LienzoMultipleSelectionControlTest.java [hunk 1]
```diff
@@ -241,6 +246,42 @@ public void testSelect() {
...
+                mock(LienzoMultipleSelectionControl.CursoredSelectionShapeProvider.class);
+        tested = new LienzoMultipleSelectionControl<>(selectionControl,
+                                                      canvasSelectionEvent,
+                                                      clearSelectionEvent,
+                                                      ssp);
+
+        tested.init(canvasHandler);
+        when(selectionManager.getSelectedItems()).thenReturn(selectedItems);
+        when(selectionManager.getSelectedItems().isSelectionGroup()).thenReturn(true);
+        tested.onCanvasSelection(new CanvasSelectionEvent(canvasHandler, Arrays.asList(ELEMENT_UUID,ELEMENT_UUID_2,ELEMENT_UUID_3,ELEMENT_UUID_4,ELEMENT_UUID_5)));
```
Comment (LuboTerifaj) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-lienzo/src/test/java/org/kie/workbench/common/stunner/client/lienzo/canvas/controls/LienzoMultipleSelectionControlTest.java]: Can you please use kie code style formatting here?

**Comment by LuboTerifaj** [comment]
Hi @handreyrc ,

I have found one more issue:
1. Ensure the "Explore diagram" is open.
2. Create any container (sub-process or lane) with any node inside it.
3. Deselect any node by clicking on canvas.
4. Close the "Explore diagram" by switching to "Properties panel" or by closing it.
5. Open back "Explore diagram".
**Actual result:**
The container is selected. If the process contains more than one container, the last created is selected.
**Expected result:**
The process is selected (the first item in the list in "Explorer").

The issue also causes that if the multiple nodes were selected before, the selection is lost.

**Comment by romartin** [comment]
Hey @LuboTerifaj @handreyrc 

First of all thanks @LuboTerifaj for the great testing, good catches! In fact, most of the issues that you describe are quite "common", I mean, I use to face any of those by doing common things, so although maybe most are not introdyuced by this ticket, I think it would be really nice fixing all those in a singe run, otherwise, if we start opening tickets, those probably will take more time to be fixed...

Sooo IMO, rather than the great job done by @handreyrc , I would continue by fixing the issues found. Maybe an option is to update title and description for [JBPM-8901](https://issues.redhat.com/browse/JBPM-8901) and include all those issue in the scope.

WDYT?

Thanks!

**Comment by romartin** [comment]
One note about the issue on the "explorer panel" (see [comment](https://github.com/kiegroup/kie-wb-common/pull/3108#issuecomment-577245884)) - it's not really a bug, although I also think we should change this behavior as @LuboTerifaj said...

The thing is that, once opening the explorer, by default it selects some node in the tree, and by selecting it implies it's being also selected on the canvas. So a potential fix would be just behaving as:
* If a single node selected on the canvas, once opening the explorer panel, select it on the tree as well
* If multiple nodes or no node selected on the canvas, once opening the explorer panel, do not select nothing on the tree, so the canvas selection will not change

**Comment by handreyrc** [comment]
Hey @LuboTerifaj , @romartin : I've tested the new issues reported and they are triggered from different components and their interactions. 
I totaly agree, and I don't see any problems in incorporating these issues to the original ticket. I'll update title and description for [https://issues.redhat.com/browse/JBPM-8901](url) and include all those issues into the scope.

**Comment by romartin** [comment]
Hey @handreyrc !

Just asking - how are you doing with these issues? Already back to work on it? Just curious because we're releasing soon, and if possible, would be good introducing these fixes for next release as well.
So no worries, just please give us some status on this. 

Thanks!

**Comment by handreyrc** [comment]
Hey @romartin, I'm already working on them.  

The issues below are already fixed:
1)If user selects multiple nodes on canvas (blue rectangle around them) and then select single node in the navigation dock, the group selection will not disappear.
2) After the nodes are moved, a different (mouse) pointer is being shown.
3) When multiple nodes are selected and a node with resize button (any activity) was selected before, this resize button is still displayed. It shouldn't be displayed since you can not interact with it.
4) The toolbox menu is not displayed when selecting a single component in the group selection.

Still working on: 
5) Clicking on Text Annotation node which is inside the selected group does not deselect multiple selection.
6) It is not possible to select associations and sequence flows when they are part of multiple selection group.
7) If the process contains more than one container, the last created is selected.

I think I won't be able to deliver all of them prior to the release.

**Comment by mbiarnes** [comment]
Jenkins execute full downstream build

**Comment by romartin** [comment]
Hey @handreyrc 

Thanks for your work and the [summary above](https://github.com/kiegroup/kie-wb-common/pull/3108#issuecomment-580517937)! :+1: 

Maybe we can try to introduce in the next release the issues you already fixed, and report as a new ticket the ones still missing, if that works for you, in order to introduce, at least, a few more bug fixes on the next release, instead of not introducing yet any of those... Anyway this also depends on the available time for QE to properly test again this PR, but I think it worths doing some small effort trying to split and do this... WDYT @LuboTerifaj @lazarotti @handreyrc ?

Thanks!

**Comment by handreyrc** [comment]
@romartin @LuboTerifaj @lazarotti : The issues 5 and 6 above are pretty much the same. When a multi-selection is on-going, the selection rectangle is drawn on top of associations and text annotations layers preventing the user from selecting them. I've put some effort on it and I couldn't find a quick fix for these issues yet.
If all of you agree, I can update the PR with the 4 issues fixed and also update the original Issue description.   We could split this issue with no restrictions of my part!

**Comment by romartin** [comment]
that works for me @handreyrc , but we should confirm with @LuboTerifaj is QE will have time to take a look at it yet. WDYT?

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by LuboTerifaj** [comment]
Hello @romartin , @handreyrc ,

I agree with splitting the issue. I can have a look at the PR once the FDB is green and possibly report unresolved issues.

However, I cannot promise it will be done before cut-off since there are more important jiras for the release.

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by romartin** [comment]
Ok thanks @LuboTerifaj , I think it worth trying it, work is already done, so if @handreyrc is agree too, let's split! :+1: 

@handreyrc can you manage that? I mean, creating a new ticket (so will end up in a new PR) and fixing the builds on this PR so @LuboTerifaj can test. 

Thx!

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
@romartin @LuboTerifaj : Hey I created a new issue for the Text Annotation and association multi-selection issues once they are pretty similar, [JBPM-9018](https://issues.redhat.com/browse/JBPM-9018). Please feel free to add more information if needed.
Regarding to  the issue on the "explorer panel", as @romartin  said above, it may not be an issue. However we can change it as suggested. Should I create a new task or a bug for this item?

**Comment by handreyrc** [comment]
Jenkins retest please

**Comment by handreyrc** [comment]
Jenkins execute full downstream build

**Comment by handreyrc** [comment]
@romartin @LuboTerifaj: Could you review this PR please? I already updated the Jira too [JBPM-8901](https://issues.redhat.com/browse/JBPM-8901).
Regarding to the SonarCloud coverage issues. I did the best iI could to maximize the code coverage making sure the methods changed are being consumed by the tests. Due to the nature of the code I don't think I can improve it much more than that.

**Comment by jomarko** [comment]
@LuboTerifaj thank you for the issues you reported. I can confirm JBPM-9018 is issue also for DMN editor. The last two are not as DMN editor does not have Explorer.

**Pull request merged**